src/order-signing/utils.ts

```diff
@@ -230,3 +234,27 @@ export function getDomain(chainId: SupportedChainId): TypedDataDomain {
 
   return domainGp(chainId, settlementContract)
 }
+
+/**
+ * Generate a deterministic order ID for the specified order.
+ * @param {SupportedChainId} chainId The chain Id
+ * @param {Order} order order to sign
+ * @param {Pick<OrderUidParams, 'owner'>} params order unique identifier parameters.
+ */
+export async function generateOrderId(
```

Comment:
I would swear we had already one piece of code doing this. I remember in the watch-tower and SDK deriving the orderId form the order data

Reply:
This is what I meant:
https://github.com/cowprotocol/cow-sdk/blob/97d03154bbc8ef47c218bef57dbf518f41dd3a10/src/composable/ConditionalOrder.ts#L286